lua/nvim_lsp/configs.lua

```diff
@@ -34,7 +34,7 @@ function configs.__newindex(t, config_name, config_def)
   -- which is why this is a function, so that it can refer to the settings
   -- object on the server.
   local function add_handlers(config)
-    assert(not lsp.callbacks, "lsp.callbacks has been deprecated. See here for more: https://github.com/neovim/neovim/pull/12655")
+    assert(not config.callbacks, "lsp.callbacks has been deprecated. See here for more: https://github.com/neovim/neovim/pull/12655")
```

Comment:
I tested JDT LS using the provided config on master and the proposed config in this pull. Using this `assert` it fails on `master` and works for `patch-1` :relieved: 

As for `lsp.callbacks`, I tried the following:

```sh
nvim -u NORC -c 'lua assert(not vim.lsp.callbacks, "Failed!")'
```

and it failed. So I replaced the `lsp` reference with `config`.

Reply:
ah, yeah, that's what I mean to say. I wanted you to assert about the config (which is the thing the user can control).